database/db_users.py

```diff
@@ -86,8 +90,23 @@ def update_general_user_info(self, user_dict):
         IndexError - invalid input format
         UserDoesNotExists - invalid user_id
         """
-        # TODO: check if input data is correct
-        pass
+        if not {'user_id', 'username', 'email', 'phone_number'}.issubset(user_dict):
+            raise IndexError
+
+        # Check user existence
+        self.get_general_user_info(user_dict['user_id'])
+
+        # Check if provided email is unique
+        try:
+            some_user_id = self._get_user_id_by_email(user_dict['email'])
+            if some_user_id != user_dict['user_id']:
+                raise EmailIsNotUnique
+        except UserDoesNotExists:
+            pass
+
+        for key in ['username', 'email', 'phone_number']:
+            if key:
```

Comment:
What case is this being checked here? Key is always a string from the defined list.

Reply:
Changed to: `user_dict[key]`